hepcrawl/spiders/elsevier_spider.py

```diff
@@ -0,0 +1,1079 @@
+        if arxiv_urls:
+            arxiv_id = arxiv_urls[0].split(":")[-1]
+            if arxiv_id and '.' not in arxiv_id:
+                return arxiv_id.strip("/")
+            else:
+                return u"arxiv:{}".format(arxiv_id)
+
+    def _get_ref_title(self, ref):
+        """Return a references title (and possible translated title)."""
+        title = self._fix_node_text(ref.xpath(
+            ".//sb:contribution/sb:title/sb:maintitle//text()").extract())
+        trans_title = ref.xpath(
+            ".//sb:contribution/sb:translated-title/sb:maintitle//text()"
+        ).extract()
+        if title and trans_title:
+            # NOTE: concatenating title with translated title:
+            title = "{} ({})".format(title, self._fix_node_text(trans_title))
+        elif trans_title:  # NOTE: is this case even possible?
+            title = trans_title
+
+        return unicode(title)
+
+    @staticmethod
+    def _get_ref_journal_title(ref):
+        """Return a journal title. Treats book series as a journal."""
+        journal_title = ''
+        if ref.xpath(".//sb:issue"):
+            journal_title = ref.xpath(
+                ".//sb:issue//sb:maintitle/text()").extract()
+            # NOTE: this is handling special issue titles, better alternatives?
+            journal_title = "; ".join(journal_title)
+        elif ref.xpath(".//sb:edited-book") and ref.xpath(".//sb:book-series"):
+            journal_title = ref.xpath(
+                ".//sb:book-series//sb:maintitle/text()").extract_first()
+        elif ref.xpath(".//sb:book") and ref.xpath(".//sb:book-series"):
+            journal_title = ref.xpath(
+                ".//sb:book-series//sb:maintitle/text()").extract_first()
+
+        return journal_title
+
+    @staticmethod
+    def _get_ref_book_title(ref, title):
+        """Return a book title."""
+        book_title = ''
+        if ref.xpath(".//sb:book") and ref.xpath(".//sb:book-series"):
+            book_title = ref.xpath(
+                ".//sb:book//sb:maintitle/text()").extract_first()
+        elif ref.xpath(".//sb:book"):
+            book_title = title
+            if not book_title:
+                book_title = ref.xpath(
+                    ".//sb:book//sb:maintitle/text()").extract_first()
+        elif ref.xpath(".//sb:edited-book"):
+            book_title = ref.xpath(
+                ".//sb:edited-book//sb:maintitle/text()").extract_first()
+            if not book_title:
+                book_title = ref.xpath(
+                    ".//sb:edited-book/sb:title/ce:inter-ref/text()"
+                ).extract_first()
+        else:
+            book_title = ref.xpath(
+                ".//sb:book//sb:maintitle/text()").extract_first()
+        return book_title
+
+    @staticmethod
+    def _fix_node_text(text_nodes):
+        """Text might be split to multiple elements, this joins them.
+
+        Input must me a list. Returns a string."""
+        # NOTE: should this be more general?
```

Comment:
@jalavik OK, `Join` can be used with titles. But for other cases I'm not sure this should be moved to the loader. I'm using `_fix_node_text` to process text that will not be the final product (eg. in `_get_ref_title`). Also it's removing unwanted whitespaces in addition to joining.